rasa/core/processor.py

```diff
@@ -184,12 +185,26 @@ async def _update_tracker_session(
                 f"Starting a new session for conversation ID '{tracker.sender_id}'."
             )
 
+            action_session_start = self._get_action(ACTION_SESSION_START_NAME)
+            # TODO: Remove in 3.0.0 and describe migration to `session_start_metadata`
+            # slot in migration guide.
+            if isinstance(
+                action_session_start, rasa.core.actions.action.ActionSessionStart
```

Comment:
This just determines that action_session_start has not been overriden by a custom action right?

Reply:
yes because in that case it doesn't make sense to add metadata